lib/lz4frame_static.h

```diff
@@ -82,6 +82,60 @@ typedef enum { LZ4F_LIST_ERRORS(LZ4F_GENERATE_ENUM) } LZ4F_errorCodes;
 LZ4F_errorCodes LZ4F_getErrorCode(size_t functionResult);
 
 
+
+/**********************************
+ *  Bulk processing dictionary API
+ *********************************/
+typedef struct LZ4F_CDict_s LZ4F_CDict;
+
+/*! LZ4_createCDict() :
+ *  When compressing multiple messages / blocks with the same dictionary, it's recommended to load it just once.
+ *  LZ4_createCDict() will create a digested dictionary, ready to start future compression operations without startup delay.
+ *  LZ4_CDict can be created once and shared by multiple threads concurrently, since its usage is read-only.
+ * `dictBuffer` can be released after LZ4_CDict creation, since its content is copied within CDict */
+LZ4F_CDict* LZ4F_createCDict(const void* dictBuffer, size_t dictSize);
+void        LZ4F_freeCDict(LZ4F_CDict* CDict);
```

Comment:
note : I was considering to re-use existing lz4frame API design, hence have `LZ4F_freeCDict()` generate an `LZ4F_errorCode_t` return code even if it's not useful and invite users to do more work, just to keep it equivalent to existing `LZ4F_freeCompressionContext()` and `LZ4F_freeDecompressionContext()`. 

But then I realised that even `LZ4F_createCDict()` is different : 
its return value is a pointer to the target object `LZ4F_CDict*`, or `NULL` if there is an error : 
This is different from `LZ4F_createCompressionContext()`, which returns an `LZ4F_errorCode_t`, and requires its first argument to be a mutable pointer to target object `LZ4F_CDict**`. 
This design feels inferior, as it's more complex to read (and gets one step closer to [triple-start programmer status](http://wiki.c2.com/?ThreeStarProgrammer)), but also because it's not possible to `const` the resulting pointer, which I consider valuable these days.

I'm now wondering if it makes any sense to half-adapt `CDict` API only on the `free` side.

Alternatively, the other option is to refactor the entire `LZ4F_createCDict()/LZ4F_freeCDict()` API to look like `LZ4F_createCompressionContext()/LZ4F_freeCompressionContext()`. But that feels less and less beneficial...

To sum up : 
```
LZ4F_CDict* const cdict = LZ4F_createCDict(...);
if (cdict==NULL) {
```
vs
```
LZ4F_CDict* cdict; 
LZ4F_errorCode_t createError = LZ4_createCDict(&cdict, ...);
if (LZ4F_isError(createError)) {
```